data_models/contents.py

```diff
@@ -32,6 +32,7 @@ class _Content:
 
     def __init__(self, content_type):
         self.content_type = content_type
+        self.content_index = 0  # default value
```

Comment:
It can start from 0 too. The absolute values don't matter and its only the relative (i.e absolute difference) between indices that matter. I've initialized it as 0 to the default value until i initialize the field with the correct value.

Reply:
In that case, initializing with -1 will be better. To clearly show that the value is currently invalid.

Reply:
Updated accordingly!